### DISCUSSION TITLE
Would the TOC extension work on Retro?

### DISCUSSION THREAD

**Discussion by ew-meetup** [jupyterlite/jupyterlite#985] [Q&A] [answered]
I would like to have the [toc2 extension](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/toc2/README.html) (or something equivalent) available in the Retro notebooks. Is this something that's possible? 

I realize that Retro seems to be developed in a separate repository, so I hope this is the correct place to ask?

I tried installing it and the jupyter lab equivalent, but haven't had any success in seeing the navigation in the Retro view.

<img width="578" alt="image" src="https://user-images.githubusercontent.com/54481302/220857248-ce4f28d2-d556-4e16-829b-b5ea747df268.png">

**Comment by jtpio** [comment, marked as answer]
Thanks @ew-meetup.

The Retro interface is provided by [RetroLab](https://github.com/jupyterlab/retrolab) which is the precursor of [Notebook 7](https://jupyter.org/enhancement-proposals/79-notebook-v7/notebook-v7.html). And development is now fully focused on Notebook 7.

Notebook 7 now supports using the Table of Contents since [7.0.0a7](https://github.com/jupyter/notebook/releases/tag/v7.0.0a7):

![image](https://user-images.githubusercontent.com/591645/195544813-22e7dec9-846f-4aaa-913a-36a9ed908036.png)

If you want to give it a try with a normal Jupyter (not JupyterLite) you can try the pre-release with `pip install --pre notebook`.

Which means the table of contents extension should automatically be supported when we update JupyterLite to use the JupyterLab 4 and Notebook 7 packages: https://github.com/jupyterlite/jupyterlite/issues/826. This will hopefully happen soon since JupyterLab 4 will enter beta in the coming weeks.
